build-tools/packages/build-infrastructure/src/test/fluidRepo.test.ts

```diff
@@ -104,3 +108,24 @@ describe("loadFluidRepo", () => {
 		});
 	});
 });
+
+describe("setDependencyRange", () => {
+	const repo = loadFluidRepo(testRepoRoot);
+	const main = repo.releaseGroups.get("main" as ReleaseGroupName);
+	assert(main !== undefined);
+
+	const group2 = repo.releaseGroups.get("group2" as ReleaseGroupName);
+	assert(group2 !== undefined);
+
+	afterEach(async () => {
+		await git.checkout(["HEAD", "--", testRepoRoot]);
+		repo.reload();
+	});
+
+	it("updates the dependency range", async () => {
+		// eslint-disable-next-line @typescript-eslint/no-non-null-assertion , @typescript-eslint/no-unsafe-call
+		await setDependencyRange(main.packages, group2.packages, semver.parse(group2.version)!);
```

Comment:
Pull the semver.parse out and assign it to a variable, then assert the variable is not null (or check that it isn't).